crates/nu-cli/src/repl.rs

```diff
@@ -223,43 +210,45 @@ fn get_line_editor(
     Ok(line_editor)
 }
 
-///
+struct LoopContext<'a> {
+    engine_state: &'a mut EngineState,
+    stack: &'a mut Stack,
+    line_editor: Reedline,
+    nu_prompt: &'a mut NushellPrompt,
+    temp_file: &'a Path,
+    use_color: bool,
+    entry_num: &'a mut usize,
+    nushell_path: &'a str,
+}
+
 /// Perform one iteration of the REPL loop
 /// Result is bool: continue loop, current reedline
 #[inline]
-fn loop_iteration(
-    engine_state: &mut EngineState,
-    stack: &mut Stack,
-    line_editor: Reedline,
-    nu_prompt: &mut NushellPrompt,
-    temp_file: &Path,
-    use_color: bool,
-    entry_num: &mut usize,
-) -> Result<(bool, Reedline)> {
+fn loop_iteration(ctx: LoopContext) -> (bool, Reedline) {
     use nu_cmd_base::hook;
     use reedline::Signal;
     let loop_start_time = std::time::Instant::now();
```

Comment:
This might a little bit of bikeshedding, but you could destructure the `LoopContext` here:
```rust
let LoopContext {
    engine_state,
    stack,
    // ...
} = ctx;
```

This should reduce some of the smaller changes sprinkled about.

Reply:
done!